### PR TITLE
Add inc update field tests

### PR DESCRIPTION
**Pull request by PatersonProjects**
This PR adds the tests for the $inc field operator, with tests matching the structure of the existing min and max operators where applicable. Tests include BSON coverage, field handling, and the $inc specific type conversion as fields are incremented, as well as edge cases and special values.

Ref: Issue #38, #393

### CODE REVIEW

File: documentdb_tests/compatibility/tests/core/operator/update/fields/inc/test_inc_update_methods.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+    result = execute_command(
+        collection, {"find": collection.name, "filter": {"_id": test.expected["_id"]}}
+    )
+    assertSuccess(result, [test.expected], msg=test.msg)
+
+
+# Property [findAndModify Context]: $inc via findAndModify with new:true returns the
+# post-increment document.
+def test_inc_find_and_modify_returns_after(collection):
```
Comment (eerxuan) [inline documentdb_tests/compatibility/tests/core/operator/update/fields/inc/test_inc_update_methods.py]: drop test_inc_find_and_modify_returns_after from test_inc_update_methods.py.

  The findAndModify wiring for update operators belongs (or should belong, if the repo doesn't have it yet) in
  core/query_and_write/commands/findAndModify/test_smoke_findAndModify.py parameterized over operators — the same way per-stage operator wiring works.

**Review by eerxuan — COMMENTED** [review]
[Required] Missing test_inc_array_positional.py

  Both sibling files exist:
  - min/test_min_array_positional.py (75 lines)
  - max/test_max_array_positional.py

  These verify the operator wires to the positional operators ($, $[], $[<identifier>]). Per the file's own docstring:

  ▎ One representative case per positional operator ($, $[], $[elem]) to confirm $min accepts them. Exhaustive positional operator coverage lives in update/array/positional*/.

File: documentdb_tests/compatibility/tests/core/operator/update/fields/inc/test_inc_update_methods.py (hunk 1 above)
Comment (PatersonProjects) [reply documentdb_tests/compatibility/tests/core/operator/update/fields/inc/test_inc_update_methods.py]: Dropped test, noted

**Review by eerxuan — APPROVED** [review]
